### PR TITLE
#5 WordPress plugin uses Password generation that does not always conform to our Strong password setting

### PR DESCRIPTION
**Pull request by staggel**
Github Issue :#5

### CODE REVIEW

File: utils/utils.php [hunk 1]
```diff
@@ -618,13 +618,9 @@ function tlms_recordLog($message){
 if(!function_exists('tlms_passgen')){
     function tlms_passgen($length = 8){
 
-        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
-        $randomString = '';
-        for ($i = 0; $i < $length; $i++) {
-            $randomString .= $characters[rand(0, strlen($characters) - 1)];
-        }
+		$length = max($length,8);
```
Comment (Bellangelo) [inline utils/utils.php:626]: Remove this line or the 620 line.

File: utils/utils.php [hunk 2]
```diff
@@ -618,13 +618,9 @@ function tlms_recordLog($message){
...
```
Comment (Bellangelo) [inline utils/utils.php]: Use spaces.

File: utils/utils.php [hunk 3]
```diff
@@ -618,13 +618,9 @@ function tlms_recordLog($message){
...
 
-        return $randomString;
+        return wp_generate_password($length) . 'aA1'; //aA1 ensures that password has one uppercase letter, one lowercase and one digit
```
Comment (Bellangelo) [inline utils/utils.php]: Add a space after the `//`.

**Review by Bellangelo — CHANGES_REQUESTED** [review]

File: utils/utils.php (hunk 1 above)
Comment (staggel) [reply utils/utils.php:626]: 77d9fab

File: utils/utils.php (hunk 2 above)
Comment (staggel) [reply utils/utils.php]: 77d9fab

File: utils/utils.php (hunk 3 above)
Comment (staggel) [reply utils/utils.php]: 77d9fab

**Review by Bellangelo — APPROVED** [review]

File: utils/utils.php [hunk 4]
```diff
@@ -618,13 +618,8 @@ function tlms_recordLog($message){
...
 
-        $characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
-        $randomString = '';
-        for ($i = 0; $i < $length; $i++) {
-            $randomString .= $characters[rand(0, strlen($characters) - 1)];
-        }
-
-        return $randomString;
+        $length = max($length, 8);
+        return wp_generate_password($length) . 'aA1'; // aA1 ensures that password has one uppercase letter, one lowercase and one digit
```
Comment (ypanagopoulos) [inline utils/utils.php]: You can make aA1 random by creating 3 pools of characters
a-z, A-Z, 0-9 and adding a single character from each pool.

At the end you can pass the string from str_shuffle() to make sure that those characters are not always positioned at the end of the string

**Review by ypanagopoulos — CHANGES_REQUESTED** [review]

File: utils/utils.php (hunk 4 above)
Comment (staggel) [reply utils/utils.php]: [4b5578b](https://github.com/epignosis/talentlms-wordpress/pull/6/commits/4b5578b3b6fdead109e926493799f2840b5711e9)

**Pull request merged**